### ISSUE TITLE
require('react-css-modules') breaks webpack source-maps

### ISSUE THREAD

**Issue by sunyang713** [gajus/react-css-modules#115]
Simply requiring the module, even without doing anything, causes webpack's source-map feature to search for `linkClass.js.map` which won't be found. That is, when I open my browser, I get a 404 because `linkClass.js.map` is not found. ~~Digging into the compiled bundle shows
`//# sourceMappingURL=linkClass.js.map`
directly after the react component.~~ Edit: this isn't relevant.

This is caused by something as simple as:
```javascript
import React from 'react'
import CSSModules from 'react-css-modules'

export default function MyComp(props) {
  return (
    <div />
  )
}
```

Any thoughts? It's sort of hard to explain clearly, so let me know if what I'm saying doesn't make sense.

**Comment by gajus** [comment]
<img width="362" alt="screen shot 2016-03-23 at 16 27 33" src="https://cloud.githubusercontent.com/assets/973543/13988341/2d5aeb9a-f114-11e5-9013-ac16a17f3d74.png">

`*.js.map` files do exist. There should be no problem with resolving them.

Please share your webpack configuration.

**Comment by sunyang713** [comment]
This isn't the entire file, but I think these are the (possibly) important parts towards this problem:

```javascript
module.exports = {
  devtool: "#source-map",
  entry: path.resolve(srcPath, "index"),
  output: {
    filename: "[name].bundle.js",
    path: path.resolve("dist"),
    pathinfo: true,
    publicPath: "/"
  },
  module: {
    loaders: [{
      test: /\.css$/,
      include: path.resolve(assetsPath, "css"),
      loaders: [
        "style",
        "css",
        "postcss"
      ]
    }, {
      test: /\.styl$/,
      include: path.resolve(assetsPath, "styles"),
      loaders: [
        "style?sourceMap",
        "css?modules&importLoaders=1&localIdentName=[path]___[name]__[local]___[hash:base64:5]",
        "postcss",
        "stylus?sourceMap"
      ]
    }]
  }
}

```

Sorry if I simply don't understand the sourcemap functions of webpack well. What confuses me is, yes the `*.js.map` files exist in the `dist/` of `react-css-modules`, but that shouldn't have to do anything with my own webpack bundle right? My webpack config produces a `main.bundle.js`, and a `main.bundle.js.map`. My browser *should* always be looking for `main.bundle.js.map`, but somehow the webpack compilation inserts `//# sourceMappingURL=linkClass.js.map` into my build, which causes my browser to search for `linkClass.js.map`.

**Comment by sunyang713** [comment]
An update: okay, so I think this might have been my fault. I am no longer getting that error, and I think it might have had to do with how I was trying to implement SSR. I made a small change, and it looks like everything is fine.

**Comment by gajus** [comment]
> I made a small change, and it looks like everything is fine.

@sunyang713 please share your findings for the reference of others who might come across a similar issue.

**Comment by sunyang713** [comment]
Okay, I'll try. the 'small change' I made before wasn't quite right, and eventually the problem came back. Now my problem is fixed, but I honestly don't know what specific change I made fixed it. I have a hunch at the general idea, so I'll try to explain it. 

I was implementing SSR. I used an express server with webpack-dev-middleware and webpack-hot-middleware, followed by a custom 'serve' middleware based on react-router's [guide](https://github.com/reactjs/react-router/blob/master/docs/guides/ServerRendering.md) on SSR. So the webpack middleware served the files (from memory) for every HTTP request. If the file can't be found (i.e., the index.html), the middleware simple passes responsibility to the next middleware. In this case, that's the custom 'serve' middleware. Now 'serve' renders the react application to a string and sends an HTML file which contains the rendered app and the script tags to my javascript bundle (created by webpack). The client would then make new requests for any of those javascript bundles and any other found URLs.

 This is my currently working system. I believe my original issue had to do with either the order of the middleware (serve files first, then handle SSR. If you are serving the index.html from memory, I've got a sweet solution for that, just poke me and I'll explain) or me incorrectly following the react-router guide for SSR.
